devito/finite_differences/differentiable.py

```diff
@@ -157,30 +147,39 @@ def __getattr__(self, name):
         raise AttributeError("%r object has no attribute %r" % (self.__class__, name))
 
     # Override SymPy arithmetic operators
+    @call_highest_priority('__radd__')
     def __add__(self, other):
         return Add(self, other)
 
+    @call_highest_priority('__add__')
     def __iadd__(self, other):
         return Add(self, other)
 
+    @call_highest_priority('__add__')
     def __radd__(self, other):
         return Add(other, self)
 
+    @call_highest_priority('__rsub__')
     def __sub__(self, other):
         return Add(self, -other)
 
+    @call_highest_priority('__sub__')
     def __isub__(self, other):
         return Add(self, -other)
 
+    @call_highest_priority('__sub__')
     def __rsub__(self, other):
         return Add(other, -self)
 
+    @call_highest_priority('__rmul__')
     def __mul__(self, other):
         return Mul(self, other)
 
+    @call_highest_priority('__mul__')
     def __imul__(self, other):
         return Mul(self, other)
 
+    @call_highest_priority('__mul__')
     def __rmul__(self, other):
         return Mul(other, self)
 
```

Comment:
missing `call_highest_priority` for `__pow__`?

Reply:
Don't think so, `pow` is a bi weird, for like `a + b` it's fine for for `a^b` what would be the opposite? There is not `rpow` or concept of it.